includes/namespace.php

```diff
@@ -87,11 +87,15 @@ function rewrite_flush() {
  */
 function activate( $network_wide = false ) {
 	$network_wide = (bool) $network_wide;
+
+	// Bootstrap the plugin upon activation.
+	PluginFactory::create()->register();
```

Comment:
The `ServiceBasedPlugin::activate()` should already contain a call to `ServiceBasedPlugin::registerServices()`, so this should not be needed.

Is the problem with the `setup_new_site()` ... ?

Reply:
Yeah there was an error when activating the plugin via WP-CLI, see the backtracke here: https://github.com/google/web-stories-wp/runs/3422113347#step:11:715

I think an alternative solution would be to move the `setup_new_site()` call after the `PluginFactory::create()->activate()` call

Reply:
Why would activating the plugin "setup a new site"? I don't fully grasp what is happening here and assume this could be refactored into something that "just works"...

I just looked into the `setup_new_site`, and noticed a few things:
- the rewrite rules are being flushed, but this also happens as part of `ServiceBasedPlugin::activate()`, so you seem to be flushing twice on activation
- the rest of the code can be either removed (registration is part of activation), or added as activation events.

So, I think that if you add `Activatable` to `User\Capabilities` and `Database_Upgrader` and call the needed methods, you can remove `setup_new_site()` entirely.

Reply:
For the `Database_Upgrader` service, it is just a matter of adapting the `'admin_init'` registration action to something more permissive and then adding more detailed checks as `Conditional`.

Reply:
I also see thet `setup_new_site()` is used 'wp_initialize_site', which explains the function name.

Again, I think most of this can be simplified away via a better use of the services flow.

Reply:
Totally makes sense. I don't know why we aren't even using the `Activateable` and `Deactivateable` interfaces at all right now... I'll give it a try.

Reply:
Do you think it would make sense to create a new interface for hooking into `wp_initialize_site`?

Reply:
Yes, and I think we already discussed this before somewhere, let me see if I can find that...

Reply:
See https://github.com/google/web-stories-wp/pull/7494

Reply:
Not sure why that was just closed, might that have been by accident? Looks like it was fully on the right track... Maybe it was too difficult to get the existing tests to pass with that change...?